### DISCUSSION TITLE
[Feature Request] Riivolution Support

### DISCUSSION THREAD

**Discussion by ExplosiveT** [Aurelio92/RVLoader#76] [Ideas]
make Riivolution xml files work in RVLoader

**Comment by NikyPSRZ** [comment]
Not only the .xml files, remember that literally every mod that uses Riivolution needs a folder with the mod's files.

**Comment by DraconicNEO** [comment]
Yeah it would be great if we had mod support in RVloader without needing to Patch ISOs manually. It's incredibly janky and archaic. We do have the [source code for Riivolution](https://github.com/AerialX/rawksd) which was released by one of the devs so it wouldn't be needed to completely reinvent the patching process.

I get that some people would become hot and bothered by people being able to run mods without discs but honestly, that's a really silly attitude to have in 2026, and especially doesn't make sense in the context of portables which almost never have disc drives.

**Reply by NikyPSRZ** [reply]
And bulders like Riivolution ISO Builder doesn't even work, because RVLoader is highly strict when booting and identificating a game, or, anything you want to boot. Other loaders lke USB Loader GX just run anything like if they are Dolphin.

**Reply by DraconicNEO** [reply]
Unfortunately it seems like RVloader isn't actively developed anymore by Aurelio so any changes will likely have to be made in a fork. Hopefully it'll happen since there are a lot of improvement RVloader needs to be able to play most games well. Including tilt support.

At least having Riivolution mod support would allow modifying games that don't work to remove their control requirements but without that there isn't much we can do.

**Reply by Aurelio92** [reply]
I am still developing RVLoader. Riivolution support is planned, but not straightforward (as gct cheats were). In order for it to work properly I need to finish a side project first

**Reply by DraconicNEO** [reply]
Oh awesome. I was under the impression the project was abandoned. I'm happy to hear that it's still being worked on.

**Reply by DraconicNEO** [reply]
Something interesting which could potentially help an implementation be developed faster. Someone recently released a fork of Riivolution with USB-loading support called Friivolution. Maybe this could help us get modding support for RV loader. 

Of course I doubt it it'll work as is in portables. Most loaders don't since they don't apply the no disc drive patches for some reason.

**Reply by Bryceshaw06** [reply]
Direct forks like that are in violation of Riivolution's license. Refer to https://github.com/AerialX/rawksd/issues/7

**Reply by DraconicNEO** [reply]
> Direct forks like that are in violation of Riivolution's license. Refer to [AerialX/rawksd#7](https://github.com/AerialX/rawksd/issues/7)

Hot take. I don't think it's a a real issue these days since tueidj fell off the face of the Earth and isn't going to be sending any DMCAs or harassing people for violating his principles when it comes to backup-loading, and AerialX doesn't seem to have a problem with it. 

Although fair enough. If some people have uneasy consciences around this, that's up to them.

**Reply by DraconicNEO** [reply]
> Direct forks like that are in violation of Riivolution's license. Refer to [AerialX/rawksd#7](https://github.com/AerialX/rawksd/issues/7)

Also Riivolution's published source code has also been used to include Riivolution mod support in Dolphin emulator already. I don't think this is as much of an issue as you think it is.

**Reply by Bryceshaw06** [reply]
The dolphin source is a reimplementation, not a port. Check the code, it is substantially different. They did it that way because of the license issue I mentioned. 

And if one person breaks a license, it doesn't make it okay for other people to break it too :)

**Reply by DraconicNEO** [reply]
>The dolphin source is a reimplementation, not a port. Check the code, it is substantially different. They did it that way because of the license issue I mentioned.

I can't be sure if changes they made were because of pro-copyright beliefs on their part or because of changes that need to happen to make it run on an emulator but it's a fair enough point.

> And if one person breaks a license, it doesn't make it okay for other people to break it too :)

I would argue that it is okay to break licenses regardless of how many people break them, I am anti-copyright and against others owning ideas. The real concern for copyright is enforcement but this isn't a real issue for orphaned works but like I said in my last comment this is a subjective opinion and is off-topic for this thread.

**Comment by NikyPSRZ** [comment]
Wait, RVLoader actually has suppot for GCT cheatcodes, why Riivolution can't run here?
